_posts/2019-09-30-introducing-arrow-flight.md

```diff
@@ -0,0 +1,260 @@
+
+Flight initially is focused on optimized transport of the Arrow columnar format
+(i.e. "Arrow record batches") over [gRPC][1], Google's popular HTTP/2-based
+general-purpose RPC library and framework. While we have focused on integration
+with gRPC, as a development framework Flight is not intended to be exclusive to
+gRPC.
+
+In the upcoming 0.15.0 Apache Arrow release, we have developed Flight
+implementations in C++ (with Python bindings) and Java. These libraries are
+ready for beta users who are comfortable being on the bleeding edge while we
+continue to refine some low-level details in the Flight internals.
+
+## Motivation
+
+Many people have experienced the pain associated with accessing large datasets
+over a network. There are many different transfer protocols and tools for
+reading datasets from remote data services, such as ODBC and JDBC. Over the
+last 10 years, file-based data warehousing in formats like CSV, Avro, and
+Parquet has become popular, but this also presents challenges as raw data must
+be transferred to local hosts before being deserialized.
+
+The work we have done since the beginning of Apache Arrow holds exciting
+promise for accelerating data transport in a number of ways. The [Arrow
+columnar format][2] has key features that can help us:
+
+* It is an "on-the-wire" representation of tabular data that does not require
+  deserialization on receipt
+* Its natural mode is that of "streaming batches", larger datasets are
+  transported a batch of rows at a time (called "record batches" in Arrow
+  parlance). In this post we will talk about "data streams", these are
+  sequences of Arrow record batches using the project's binary protocol
+* The format is language-independent and now has library support in 11
+  languages and counting. For
+
+Implementations of standard protocols like ODBC generally implement their own
+custom on-wire binary protocols that must be marshalled to and from each
+library's public interface. The performance of ODBC or JDBC libraries varies
+greatly from case to case.
+
+Our design goal for Flight is to create a new protocol for data services that
+uses the Arrow columnar format as both the over-the-wire data representation as
+well as the public API presented to developers. In doing so, we reduce or
+remove the serialization costs associated with data transport and increase the
+overall efficiency of distributed data systems. Additionally, two systems that
+are already using Apache Arrow for other purposes can communicate data to each
+other with extreme efficiency.
+
+## Flight Basics
+
+The Arrow Flight libraries provide a development framework for implementing a
+service that can send and receive data streams. A Flight server supports
+several basic kinds of requests:
+
+* **ListFlights**: return a list of available data streams
+* **GetSchema**: return the schema for a data stream
+* **GetFlightInfo**: return a "query plan" for a dataset of interest, possibly
```

Comment:
Calling it a "query plan" might be a bit presomptuous? :-)

Reply:
Renamed to "access plan"